### DISCUSSION TITLE
Traffic Shaping: Cake

### DISCUSSION THREAD

**Discussion by BIGFAAT** [Freetz-NG/freetz-ng#585] [Unerfüllte Wünsche]
I suggest support for cake, together with a minimal config or web interface (take a look over OpenWRT):
https://man7.org/linux/man-pages/man8/tc-cake.8.html

**Comment by fda77** [comment]
I think (but s.o. could test) its not compatible with avm's routing "daemons" and PA, see
 - https://freetz-ng.github.io/freetz-ng/wiki/00_FAQ/FAQ.de.html#packet-accelerator-avm-pa
 - https://freetz-ng.github.io/freetz-ng/wiki/00_FAQ/FAQ.de.html#funktioniert-iptablesnatconntrack

**Reply by BIGFAAT** [reply]
Indeed, older models will struggle if they try to cake anything bigger than a 50mbit connection without hwoffload. But the newer router generation (or anything without mips) should be able to get the work done without hardware offloading since they have quite bigger cpus. (And, if im not wrong, cable/fiber devices too since running on different architectures).

Of course im open for any traffic shapping/QoS implementation, with for example iptables/nftables. If possible with dscp marking capabilities.

For me its the only thing i really dont like about fritz devices. It would really be nice if implemented. At least its already possible for the older ones under OpernWRT and smaller connections.

**Comment by fda77** [comment]
AVM devices are not "open" to iptables and other common network tools due to PA.
With Openwrt you'll have a recent kernel without this (and many orher) limitation
